src/pages/PoolDetails/PoolDetailsHeader.tsx

```diff
@@ -0,0 +1,208 @@
+import { Trans } from '@lingui/macro'
+import { ChainId, Currency } from '@uniswap/sdk-core'
+import blankTokenUrl from 'assets/svg/blank_token.svg'
+import Column from 'components/Column'
+import Row from 'components/Row'
+import { getChainInfo } from 'constants/chainInfo'
+import { chainIdToBackendName } from 'graphql/data/util'
+import { useCurrency } from 'hooks/Tokens'
+import useTokenLogoSource from 'hooks/useAssetLogoSource'
+import React from 'react'
+import { Link } from 'react-router-dom'
+import styled from 'styled-components'
+import { ClickableStyle, ThemedText } from 'theme'
+import { shortenAddress } from 'utils'
+
+import { ReversedArrowsIcon } from './icons'
+
+const HeaderColumn = styled(Column)`
+  gap: 36px;
+`
+
+const StyledLink = styled(Link)`
+  text-decoration: none;
+  ${ClickableStyle}
+`
+
+const FeeTier = styled(ThemedText.LabelMicro)`
+  background: ${({ theme }) => theme.surface2};
+  padding: 2px 6px;
+  border-radius: 4px;
+`
+
+const ToggleReverseArrows = styled(ReversedArrowsIcon)`
+  ${ClickableStyle}
+`
+
+interface Token {
+  id: string
+  symbol: string
+}
+
+interface PoolDetailsHeaderProps {
+  chainId?: number
+  poolAddress?: string
+  token0?: Token
+  token1?: Token
+  feeTier?: number
+  toggleReversed: React.DispatchWithoutAction
+}
+
+export function PoolDetailsHeader({
+  chainId,
+  poolAddress,
+  token0,
+  token1,
+  feeTier,
+  toggleReversed,
+}: PoolDetailsHeaderProps) {
+  const currencies = [useCurrency(token0?.id, chainId) ?? undefined, useCurrency(token1?.id, chainId) ?? undefined]
+  const chainName = chainIdToBackendName(chainId)
+  const origin = `/tokens/${chainName}`
+  return (
+    <HeaderColumn>
+      <Row>
+        <StyledLink to={origin}>
+          <ThemedText.BodySecondary>
+            <Trans>Explore</Trans>
+          </ThemedText.BodySecondary>
+        </StyledLink>
+        <ThemedText.BodySecondary>&nbsp;{'>'}&nbsp;</ThemedText.BodySecondary>
+        {/* TODO: When Explore Pool table is added, link directly back to it */}
+        <StyledLink to={origin}>
+          <ThemedText.BodySecondary>
+            <Trans>Pool</Trans>
+          </ThemedText.BodySecondary>
+        </StyledLink>
+        <ThemedText.BodySecondary>&nbsp;{'>'}&nbsp;</ThemedText.BodySecondary>
+        <ThemedText.BodyPrimary>
+          {token0?.symbol} / {token1?.symbol} ({shortenAddress(poolAddress)})
```

Comment:
does this have a failure case where a token is undefined? would look odd? jw

Reply:
For a token to be undefined would mean there is an upstream issue with thegraph where a valid pool address was given and the gql query did not return an error but the token data was undefined.